x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -591,32 +596,14 @@ static void failIfAllShardsFailed(EsqlExecutionInfo execInfo, List<Page> finalRe
     }
 
     void runCompute(CancellableTask task, ComputeContext context, PhysicalPlan plan, ActionListener<DriverCompletionInfo> listener) {
-        listener = ActionListener.runBefore(listener, () -> Releasables.close(context.searchContexts()));
```

Comment:
It looks like we used to always close the search contexts before running the listener, but now we only close them in case of an exception. Why is that?